src/read_clustering/variant_call.py

```diff
@@ -279,9 +279,43 @@ def get_read_positions_data(self, read_id, positions):
         """If position exists in read return data covering list of positions
         :param read_id: string read name
         :param positions: list of reference indices
-        :return: dataframe with both read_id equal to read_id and reference_index is in positions list
+        :return dataframe with both read_id equal to read_id and reference_index is in positions list
         """
         data = self.data[(self.data['read_id'] == read_id) & (self.data['reference_index'].isin(positions))]
         if data.empty:
             return False
         return data
+
+    def get_reads_covering_positions_data(self, positions, variant_sets):
+        """Return a dataframe with the probabilities for the modified variants in the given variant_set for their given
+        position in the positions list
+        :param positions: a list of any number of positions for the modified variants
+        :param variant_sets: a list of modified variants
+        :return df_plot: data frame with the probabilities for each variant at each position with corresponding read id.
+        """
+        temp_df = self.data[(self.data['reference_index'].isin(positions)) & (self.data['variants'].isin(variant_sets))]
+        plot_data = temp_df.loc[:, ['read_id', 'reference_index', 'variants', 'prob1', 'prob2']]
+        pos_n = len(positions)
+        select = plot_data.groupby(['read_id']).nunique()
+        a = select[(select['reference_index'] == pos_n)]
```

Comment:
@shreyamantri @aduranu I fixed a bug here that selected on the number of "variants" as well as position. You can see a test I created called `test_get_reads_covering_positions_data3` that helped find the error.